cluster/addons/podsecuritypolicies/unprivileged.yaml

```diff
@@ -0,0 +1,53 @@
+apiVersion: extensions/v1beta1
+kind: PodSecurityPolicy
+metadata:
+  name: unprivileged
+  annotations:
+    kubernetes.io/description: 'unprivileged allows full access to all pod
+      settings except those that grant host privileges.'
+    seccomp.security.alpha.kubernetes.io/allowedProfileNames: '*'
+  labels:
+    kubernetes.io/cluster-service: 'true'
+    addonmanager.kubernetes.io/mode: Reconcile
+spec:
+  privileged: false
+  allowPrivilegeEscalation: true
+  volumes: # All volume types except HostPath
```

Comment:
allowing direct use of all these inline volumes is not very unprivileged, is it? don't many of these let you reference arbitrary paths or mount namespaces? cc @kubernetes/sig-storage-pr-reviews

Reply:
Only hostpath volumes let you specify arbitrary paths.  Mount propagation is separately controlled via SecurityContext.Privileged.

Reply:
I meant things like nfs.path, quobyte.volume, vsphereVolume.volumePath, etc, that let a pod author target arbitrary volumes as the node

Reply:
Ah, I see what you mean.  But will restricting the volume types here cause users to break on upgrade?

Reply:
So I think the intent of this policy is to document an unprivileged policy for others but also apply it to some of our own components as above. I assume this list is here to allow others to apply the policy and have it work in most cases. Perhaps this is the wrong approach. Would it be more useful to provide the minimal PSP that would let a vanilla K8s cluster function and have users add the extra storage types and other things they need in a new policy? 

As is, it might be hard for a non-expert who wants to lock down their config to reason about what's required for core k8s and what can be safely removed if they aren't using it in their applications.

If we did that would we also need an unprivileged-gce policy for kube-dns etc. where you are currently applying this one?

Reply:
> will restricting the volume types here cause users to break on upgrade?

It is certainly possible. Tightening restrictions on a cluster without any knowledge of what workloads are running while guaranteeing no disruption is impossible. I would expect a rollout like this:
* upgraded clusters do not enable PSP by default
* upgraded clusters can enable PSP with opt-in by users
* new clusters can enable PSP by default

> Would it be more useful to provide the minimal PSP that would let a vanilla K8s cluster function and have users add the extra storage types and other things they need in a new policy?

Absolutely. It is extremely difficult to tighten default policies. I would expect something like:
* a privileged PSP that allows anything, bound to the superuser (and possibly to kubelets, since they can only create mirror pods and should be able to create a mirror pod with any combination of permissions)
* a restricted PSP that only allows "safe" inline volume types (secret, configmap, emptydir, projected, pvc, downwardAPI), no host access, no root permissions, no escalation, etc
* PSPs defined for specific use cases granted to specific service accounts for specific addons.